app/hotels/src/filter/score/ScoreFilter.js

```diff
@@ -0,0 +1,57 @@
+// @flow
+
+import * as React from 'react';
+import { View } from 'react-native';
+
+import ScorePopup from './ScorePopup';
+import FilterButton from '../FilterButton';
+import type { OnChangeFilterParams } from '../FilterParametersType';
+
+type Props = {|
+  minScore: number | null,
+  onChange: OnChangeFilterParams => void,
+|};
+
+type State = {|
+  isPopupOpen: boolean,
+|};
+
+const rating = ['pleasant 6+', 'good 7+', 'very good 8+', 'superb 9+'];
+
+export default class ScoreFilter extends React.Component<Props, State> {
+  state = {
+    isPopupOpen: false,
+  };
+
+  handlePopupToggle = () =>
+    this.setState(state => ({
+      isPopupOpen: !state.isPopupOpen,
+    }));
+
+  handleSave = (minScore: number | null) => this.props.onChange({ minScore });
+
+  getTitle = (minScore: number | null) =>
+    minScore
+      ? rating[minScore - 6] // Minimal score is 6
```

Comment:
This is a bit cryptic.

I would rename `rating` to `ratingTitles` and changed it to an object (`const ratingTitles = { 6: 'pleasant 6+', 7: 'good 7+', ... }`). Then this method could look like `return ratingTitles[minScore] || 'rating'`.